crates/ruff_linter/src/source_kind.rs

```diff
@@ -83,6 +85,79 @@ impl SourceKind {
             }
         }
     }
+
+    /// Write a diff of the transformed source file to `stdout`.
+    pub fn diff(&self, other: &Self, path: Option<&Path>) -> Result<()> {
+        let mut writer = io::stdout().lock();
```

Comment:
nit: maybe we could get the writer from the caller which could potentially allow us to write tests in terms of snapshots.